### PR TITLE
ARROW-4946: [C++] Support detection of flatbuffers without FlatbuffersConfig.cmake

### PR DESCRIPTION
**Pull request by xhochy**

### CODE REVIEW

File: run-cmake-format.py [hunk 1]
```diff
@@ -32,6 +32,7 @@
     'cpp/cmake_modules/FindBrotli.cmake',
     'cpp/cmake_modules/FindClangTools.cmake',
     'cpp/cmake_modules/FindDoubleConversion.cmake',
+    'cpp/cmake_modules/FindFlatbuffersAlt.cmake',
```
Comment (pitrou) [inline run-cmake-format.py]: I suppose `FindFlatbuffersAlt.cmake` is missing from the PR?

Comment (xhochy) [reply run-cmake-format.py]: Fixed

File: cpp/cmake_modules/ThirdpartyToolchain.cmake [hunk 2]
```diff
@@ -1482,7 +1482,23 @@ macro(build_flatbuffers)
 endmacro()
 
 if(ARROW_WITH_FLATBUFFERS)
-  resolve_dependency(Flatbuffers)
+  get_dependency_source(Flatbuffers)
+  if(Flatbuffers_SOURCE STREQUAL "AUTO")
+    find_package(Flatbuffers QUIET)
+    if(NOT Flatbuffers_FOUND)
+      find_package(FlatbuffersAlt)
```
Comment (pitrou) [inline cpp/cmake_modules/ThirdpartyToolchain.cmake]: Perhaps comment on why this is required?

**Review by pitrou — APPROVED** [review]
Works fine for me with Anaconda's flatbuffers 1.7.1 package.

File: cpp/cmake_modules/ThirdpartyToolchain.cmake (hunk 2 above)
Comment (xhochy) [reply cpp/cmake_modules/ThirdpartyToolchain.cmake]: Done

**Pull request closed without merging**